lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -2369,6 +2369,10 @@ public function getCommitOrderCalculator()
      */
     public function clear($entityName = null)
     {
+        if ($entityName !== null && !is_string($entityName)) {
+            throw new \InvalidArgumentException(sprintf('Argument 1 passed to %s() must be a string, %s given', __METHOD__, gettype($entityName)));
```

Comment:
This should be something like `throw ORMException::fromInvalidEntityName($entityName)`